docker/bitcoind/Dockerfile

```diff
@@ -1,19 +1,36 @@
-# Use the official Ubuntu 22.04 base image
+# Use the official Ubuntu 24.04 base image
 FROM ubuntu:24.04
 
 WORKDIR /app
 
-# Install necessary dependencies and add PPA
+# Install necessary dependencies
 RUN apt-get update && \
-    apt-get install -y software-properties-common && \
-    add-apt-repository ppa:luke-jr/bitcoincore && \
-    apt-get update && \
-    apt-get install -y \
-    bitcoind \
-    curl && \
-    apt-get clean && \
-    rm -rf /var/lib/apt/lists/*
-
+    apt-get install -y software-properties-common build-essential libtool autotools-dev automake pkg-config bsdmainutils python3 && \
+    apt-get install -y libevent-dev libboost-dev libsqlite3-dev && \
+    apt-get install -y libssl-dev libboost-all-dev git
+
+# Set Git with dummy user configuration
+# This configuration is needed for cherry-picking the specific commit
+RUN git config --global user.email "john@example.com" && \
+    git config --global user.name "John Doe"
+
+# Clone Bitcoin repository and checkout the commit for v28.0 tag
+RUN git clone https://github.com/bitcoin/bitcoin.git && \
+    cd bitcoin && \
+    git checkout 5de225f5c145368f70cb5f870933bcf9df6b92c8
+
+# Apply the specific pr https://github.com/bitcoin/bitcoin/pull/27446/commits
+# which includes the changes needed for configurable block time for signet
+RUN cd bitcoin && \
+    git fetch origin pull/27446/head && \
+    git cherry-pick d8434da3c14ed6723d86ef2cd266008d366e1413
```

Comment:
I don't like this. The PR can be changed and we can get rug-pulled (EDIT: not exactly since we are `cherry-pick`ing by the long SHA, but still my tinfoil-hat spider-senses are going crazy)
Can you grab that diff and put in a `30_sec_block.diff` file and then apply that diff?
This way we are totally sure that the diff will not change under us.

Reply:
Yeah, having hash assures nothing else is applied. But I like the idea of having a patch file which show what difference is being on top of the tag. Testing in local.